api_v3/services/StorageProfileService.php

```diff
@@ -130,5 +138,255 @@ public function listAction(KalturaStorageProfileFilter $filter = null, KalturaFi
+		}
+
+		self::setLastIdInCache($initialLastId, $lastId, $keysCache, $workerId);
+		self::fillMissingFileSyncsPath($lockedFileSyncs);
+
+		// build the response object
+		$result = new KalturaLockFileSyncsResponse;
+		$result->fileSyncs = KalturaFileSyncArray::fromDbArray($lockedFileSyncs, $this->getResponseProfile());
+		$result->limitReached = $limitReached;
+
+		return $result;
+
+	}
+
+	protected static function getCacheLayers()
+	{
+		// get caches
+		$keysCache = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_QUERY_CACHE_KEYS);
+		if (!$keysCache)
+		{
+			throw new KalturaAPIException(MultiCentersErrors::GET_KEYS_CACHE_FAILED);
+		}
+
+		$lockCache = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_LOCK_KEYS);
+		if (!$lockCache)
+		{
+			throw new KalturaAPIException(MultiCentersErrors::GET_LOCK_CACHE_FAILED);
+		}
+
+		return array($keysCache, $lockCache);
+	}
+
+	protected static function getMaxId($keysCache, $storageProfileId)
+	{
+		// get the max id / last id
+		$maxId = $keysCache->get(self::MAX_FILESYNC_ID_PREFIX . $storageProfileId);
+		if (!$maxId)
+		{
+			throw new KalturaAPIException(MultiCentersErrors::GET_MAX_FILESYNC_ID_FAILED, $storageProfileId);
+		}
+		$maxId -= 1000;
+		return $maxId;
+	}
+
+	protected static function buildFileSyncCriteria($filter, $storageProfileId)
+	{
+		// build the criteria
+		$fileSyncFilter = new FileSyncFilter();
+		$filter->toObject($fileSyncFilter);
+
+		$baseCriteria = new Criteria();
+		$fileSyncFilter->attachToCriteria($baseCriteria);
+
+		$baseCriteria->add(FileSyncPeer::STATUS, FileSync::FILE_SYNC_STATUS_PENDING);
+		$baseCriteria->add(FileSyncPeer::FILE_TYPE, FileSync::FILE_SYNC_FILE_TYPE_URL);
+		$baseCriteria->add(FileSyncPeer::DC, $storageProfileId, Criteria::IN);
+
+		$baseCriteria->addAscendingOrderByColumn(FileSyncPeer::ID);
+		$baseCriteria->setLimit(self::MAX_FILESYNCS_PER_CHUNK);
+
+		return $baseCriteria;
+	}
+
+	protected static function getFileSyncsChunk($baseCriteria, $lastId, $idLimit)
+	{
+		// get a chunk of file syncs
+		// Note: starting slightly before the last id, because the ids may arrive out of order in the mysql replication
+		$c = clone $baseCriteria;
+		$idCriterion = $c->getNewCriterion(FileSyncPeer::ID, $lastId, Criteria::GREATER_EQUAL);
+		$idCriterion->addAnd($c->getNewCriterion(FileSyncPeer::ID, $idLimit, Criteria::LESS_THAN));
+		$c->addAnd($idCriterion);
+
+		// Note: disabling the criteria because it accumulates more and more criterions, and the status was already explicitly added
+		//		once that bug is fixed, this can be removed
+		FileSyncPeer::setUseCriteriaFilter(false);
+		$fileSyncs = FileSyncPeer::doSelect($c, myDbHelper::getConnection(myDbHelper::DB_HELPER_CONN_PROPEL2));
+		FileSyncPeer::setUseCriteriaFilter(true);
+
+		return $fileSyncs;
+	}
+
+	protected static function updateLastId($fileSyncs, $idLimit)
```

Comment:
Since the last id is just returned here, no update is done to a member or static value - please rename the fucntion to getlastId